### PR TITLE
Add Fastly + S3 static sites post

### PR DESCRIPTION
**Pull request by sethvargo**
/cc @KFishner

### CODE REVIEW

File: source/blog/2016-01-26-serving-static-sites-with-fastly.markdown [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+---
+
+Serving Static Sites with Fastly, S3, and Middleman
+
+In November we announced our [partnership with Fastly](/blog/hashicorp-and-fastly.html) to power the new [HashiCorp releases service](https://releases.hashicorp.com/). Since then, we have expanded our use of [Fastly](https://www.fastly.com/) to front all of our static sites. You may have noticed subtle frontend and backend changes to our various websites - this post details the steps we took to migrate our static sites to Fastly.
+
+READMORE
+
+## Overview
+This post goes into detail about the specific steps we took to migrate our static sites to be fronted by Fastly. At a very high level, our new architecture is modeled as:
```
Comment (KFishner) [inline source/blog/2016-01-26-serving-static-sites-with-fastly.markdown]: These two sentences can probably be condensed into one:

"At a very high level, our new static site architecture is modeled as:"

File: source/blog/2016-01-26-serving-static-sites-with-fastly.markdown [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+- `--delete-removed` tells s3cmd to remove old files that are no longer tracked (part of sync).
+- `--guess-mime-type` tells s3cmd to guess the mime type from the _file extension_ instead of using python-magic (which is horribly inaccurate).
+- `--non-mime-magic` tells s3cmd to disable the python-magic mime detection entirely.
+- `--acl-public` tells s3cmd to make the resources public, read-only.
+- `--recursive` tells s3cmd to recurse into subdirectories and folders.
+- The first `--add-header` call sets the cache control on the resource to one year.
+- The second `--add-header` call sets a surrogate-key.
+- `$PROJECT` is a variable that is populated by Atlas with the name of the current site (like "terraform").
+
+Of special note is the `Surrogate-Key` field and cache timers. As you will see later, we tell Fastly to cache our content for a year at a time. We purge the cache using a surrogate key on deploy. You can read more about Fastly surrogate keys in [Fastly documentation for purging by surrogate keys](https://docs.fastly.com/guides/purging/setting-surrogate-key-headers-based-on-a-url). This will become clearer in a moment.
```
Comment (KFishner) [inline source/blog/2016-01-26-serving-static-sites-with-fastly.markdown]: no caps on `surrogate-key` probably

File: source/blog/2016-01-26-serving-static-sites-with-fastly.markdown [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+Serving Static Sites with Fastly, S3, and Middleman
+
+In November we announced our [partnership with Fastly](/blog/hashicorp-and-fastly.html) to power the new [HashiCorp releases service](https://releases.hashicorp.com/). Since then, we have expanded our use of [Fastly](https://www.fastly.com/) to front all of our static sites. You may have noticed subtle frontend and backend changes to our various websites - this post details the steps we took to migrate our static sites to Fastly.
+
+READMORE
+
+## Overview
+This post goes into detail about the specific steps we took to migrate our static sites to be fronted by Fastly. At a very high level, our new architecture is modeled as:
+
+![Atlas, S3, Fastly Flow Chart](/images/blog/serving-static-sites-with-fastly/arch.jpg)
```
Comment (KFishner) [inline source/blog/2016-01-26-serving-static-sites-with-fastly.markdown]: I might have missed this detail, but in the diagram there's a two-way error between S3 and Fastly. How does Fastly interact back to S3?

**Comment by KFishner** [comment]
very minor comments otherwise lgtm!

File: source/blog/2016-01-26-serving-static-sites-with-fastly.markdown [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+
+## What is a Static Site?
+At HashiCorp, we believe an important design and user experience is a key factor in the success of a project. Not only do we put significant time and effort to the command line interfaces, error messages, and output, but each project is given its own unique identity and website. Each project's website uses [Middleman](https://middlemanapp.com) which is a static website generator written in Ruby. Here are just a few of our static sites:
+
+- www.vagrantup.com
+- www.terraform.io
+- www.hashicorp.com
+
+The HashiCorp static sites have historically been hosted using a PaaS with a series of custom buildpacks. This process posed a few key problems. First, all requests to the static site first hit a Ruby web server, which then loaded a pre-built file from disk. This resulted in less than optimal performance and made global caching difficult and costly. A common complaint from customers was that our static websites were slow or inaccessible from different parts of the world. The pre-built file was built on each deploy, which led to occasional deployment issues ranging from conflicting Ruby versions, stalled pushes, or incomplete website deploys.
```
Comment (burzinpatel) [inline source/blog/2016-01-26-serving-static-sites-with-fastly.markdown]: Double use of word in same sentence: "First, all requests to the static site hit..."

File: source/blog/2016-01-26-serving-static-sites-with-fastly.markdown (hunk 2 above)
Comment (sethvargo) [inline source/blog/2016-01-26-serving-static-sites-with-fastly.markdown]: It's supposed to be caps actually

File: source/blog/2016-01-26-serving-static-sites-with-fastly.markdown (hunk 3 above)
Comment (sethvargo) [inline source/blog/2016-01-26-serving-static-sites-with-fastly.markdown]: Fastly pulls the content from S3 (S3 is the origin)

**Pull request merged**